src/parser/grammar.lalrpop

```diff
@@ -253,21 +253,23 @@ AnnotatedInfixExpr: UniTerm = {
 };
 
 Forall: Types =
-    "forall" <ids: Ident+> "." <ty: Types> => {
-        ids.into_iter().rev().fold(
+    <l: @L> "forall" <ids: Ident+> "." <ty: Types> <r: @R> => {
```

Comment:
Same reason as [this](https://github.com/tweag/nickel/pull/1185#discussion_r1138436246). Will fix.

Reply:
Okay, for some reason `Forall`s don't have a position unless I use the `<l: @L>` and `<r: @R>`.

Reply:
Then something else is wrong, somewhere. Either a usage of `Forall` doesn't have a `WithPos`, or an usage of `Types` which is using `Forall` doesn't get a `WithPos`. Finally, would my suggestion fix the problem? If you have several foralls in a row, because you use `Types` instead of `WithPos<Types` in the rule, you might lose position.

Reply:
> Finally, would my suggestion fix the problem? If you have several foralls in a row, because you use Types instead of WithPos<Types in the rule, you might lose position.

Ah, yes.


But nothing else seems to work (apart from the `<l: @L> and <r: @R>`)

Reply:
Well, once again, have you tried to substitute `<ty: Types>` with `<ty: WithPos<Types>>` ?

Reply:
Yes